src/app/api/invites/route.test.ts

```diff
@@ -0,0 +1,188 @@
+        user: { id: "1234", type: "SUPER_ADMIN" },
+        expires: "",
+      });
+
+      dbMock.user.findFirst.mockResolvedValue({
+        id: 1,
+        email: "test@test.com",
+        type: UserType.ADMIN,
+        passwordHash: "abc",
+      });
+
+      const formData = new FormData();
+      formData.append("email", "test@test.com");
+      formData.append("userType", "ADMIN");
+      const res = await fetch({ method: "POST", body: formData });
+      expect(res.status).toBe(409);
+    },
+  });
+});
+
+test("test email html", async () => {
+  await testApiHandler({
+    appHandler,
+    async test({ fetch }) {
+      authMock.mockReturnValueOnce({
+        user: { id: "1234", type: "SUPER_ADMIN" },
+        expires: "",
+      });
+
+      const uuidMock = jest.spyOn(uuid, "v4");
+      const mockToken = "mocked-uuid-token";
+      // @ts-expect-error: jest cannot deal with overloaded functions
+      uuidMock.mockReturnValueOnce(mockToken);
+
+      const sendEmailMock = jest.spyOn(emailModule, "sendEmail");
+
+      const formData = new FormData();
+      formData.append("email", "test@test.com");
+      formData.append("userType", "ADMIN");
+      const res = await fetch({ method: "POST", body: formData });
+      expect(res.status).toBe(200);
+
+      const emailHtml = sendEmailMock.mock.calls[0][2];
+      const urlRegex = new RegExp(`register\\?token=${mockToken}`);
+      expect(emailHtml).toMatch(urlRegex);
+    },
+  });
+});
+
+test("UserInvite expires in one day", async () => {
+  await testApiHandler({
+    appHandler,
+    async test({ fetch }) {
+      authMock.mockReturnValueOnce({
+        user: { id: "1234", type: "SUPER_ADMIN" },
+        expires: "",
+      });
+
+      const formData = new FormData();
+      formData.append("email", "test@test.com");
+      formData.append("userType", "ADMIN");
+      const res = await fetch({ method: "POST", body: formData });
+      expect(res.status).toBe(200);
+
+      const createdInvite = dbMock.userInvite.create.mock.calls[0][0].data;
+      const expirationDate = new Date(createdInvite.expiration);
+      const currentDate = new Date();
+      const oneDayInMilliseconds = 24 * 60 * 60 * 1000;
+
+      expect(expirationDate.getTime() - currentDate.getTime()).toBeCloseTo(
+        oneDayInMilliseconds,
+        -2
+      );
+    },
+  });
+});
+
+test("show email", async () => {
+  await testApiHandler({
+    appHandler,
+    async test({ fetch }) {
+      authMock.mockReturnValueOnce({
+        user: { id: "1234", type: "SUPER_ADMIN" },
+        expires: "",
+      });
+
+      const sendEmailMock = jest.spyOn(emailModule, "sendEmail");
+
+      const formData = new FormData();
+      formData.append("email", "test@test.com");
+      formData.append("userType", "ADMIN");
+      const res = await fetch({ method: "POST", body: formData });
+      expect(res.status).toBe(200);
+
+      const emailHtml = sendEmailMock.mock.calls[0][2];
+      openHtml(emailHtml);
```

Comment:
the test shouldnt open anything-- it should just verify that `sendEmail` is being called